failure-summary.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Finds failing Checks and combines each help message into an array
+ * @param  {Object} nodeData Individual "detail" object to generate help messages for
+ * @return {String}          failure messages
+ */
+var failureSummary = function failureSummary(nodeData) {
+	'use strict';
+
+	var failingChecks = {};
+	// combine "all" and "none" as messaging is the same
+	failingChecks.none = nodeData.none.concat(nodeData.all);
+	failingChecks.any = nodeData.any;
+
+	return Object.keys(failingChecks).map(function (key) {
```

Comment:
Have a look now.
Didn't like hiding away the related nodes under failure summary though.